### DISCUSSION TITLE
Rounded corner clipping

### DISCUSSION THREAD

**Discussion by AustinGraphics** [TurboWarp/extensions#886] [General]
Would it be possible to add a rounded corner function on the clipping & blending project? It would be great if you could choose the radius of the corners! 👍 @Xeltalliv 

Something like this would be cool :)
![image](https://github.com/TurboWarp/extensions/assets/66364927/50d52e51-657a-4184-8bba-3de5a2d1f218)

**Comment by yuri-kiss** [comment]
yeah i would like this so you could round images

**Reply by AustinGraphics** [reply]
Or you could make really nice buttons thag can smoothy transform like expanding and shrinking 👀

**Reply by yuri-kiss** [reply]
YES

**Comment by ghost** [comment]


**Reply by AustinGraphics** [reply]
?
why send this?
Could you delete this please it has nothing to do w the post

**Reply by yuri-kiss** [reply]
yeah its kind of just spam

**Reply by ghost** [reply]
sorry

**Comment by Xeltalliv** [comment]
> Would it be possible to add a rounded corner function on the clipping & blending project?

The short answer is no. WebGL provides a built-in feature for clipping to a rectangle (called [scissor test](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/scissor)). Before each clipped sprite is drawn, this extension enables and configures that feature and afterwards disables it. This extension just injects some new code into existing turbowarp code.

Anything other than purely a rectangle can't be done as easily and requires significant changes to how turbowarp's sprite-drawing code works. Doing that is a bad idea because other extensions also rely on that code. Or GarboMuffin may decide to change it. So less changes extension does = better compatibility.

**Reply by AustinGraphics** [reply]
Lets ignore the complexity of what im about to ask, would it be possible to use multiple 'clipping masks' to mimic rounded corners?

**Reply by yuri-kiss** [reply]
sadly the clipping masks cant be stacked 😭

**Reply by AustinGraphics** [reply]
thats really unfortunate 😔
